### PR TITLE
Add verification of SHA from `--sha`

### PR DESCRIPTION
**Pull request by ErikDanielsson**
Added verification that a commit SHA provided through `--sha` in `nf-core modules update/install` exist in the remote. 

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/modules/install.py [hunk 1]
```diff
@@ -42,6 +42,17 @@ def install(self, module):
             log.error("Cannot use '--sha' and '--prompt' at the same time!")
             return False
 
+        # Verify that a provided SHA is exist in the repo
```
Comment (drpatelh) [inline nf_core/modules/install.py:45]:
```suggestion
        # Verify that the provided SHA exists in the repo
```

File: nf_core/modules/update.py [hunk 2]
```diff
@@ -44,6 +44,21 @@ def update(self, module):
                 == "All modules"
             )
 
+        if self.prompt and self.sha is not None:
+            log.error("Cannot use '--sha' and '--prompt' at the same time!")
+            return False
+
+        # Verify that a provided SHA is exist in the repo
```
Comment (drpatelh) [inline nf_core/modules/update.py:51]:
```suggestion
        # Verify that the provided SHA exists in the repo
```

**Review by drpatelh — APPROVED** [review]
Awesome!

**Comment by ErikDanielsson** [comment]
Thanks @drpatelh!

**Pull request merged**